src/common/transformations/include/transformations/fp16_compression/mark_decompression_convert_constant_folding.hpp

```diff
@@ -47,3 +48,14 @@ class ov::pass::KeepConstAndDecompression : public MatcherPass {
     OPENVINO_RTTI("KeepConstAndDecompression", "0");
     KeepConstAndDecompression();
 };
+
+/**
+ * @ingroup ie_transformation_common_api
+ * @brief Disables ConstantFolding for Convert operation (just before MatMul operation only) and prevents conversion
+ * of f16 Consts to f32.
+ */
+class ov::pass::KeepConstAndDecompressionForMatMul : public MatcherPass {
```

Comment:
@itikhono @antonvor Don't we want to have user defined callback which will state conditions for applicable operation? Seems like better option comparing with new pass introduction?

Reply:
I propose to do this in separate PR

Reply:
Ok. Lets refactor in follow-up PR

Reply:
PR with refactoring: https://github.com/openvinotoolkit/openvino/pull/19022